src/Raven.Server/Documents/Sharding/Handlers/ShardedIncomingReplicationHandler.cs

```diff
@@ -185,6 +184,24 @@ private ReplicationBatches PrepareReplicationDataForShards(TransactionOperationC
 
             foreach (var item in dataForReplicationCommand.ReplicatedItems)
             {
+                if (item is RevisionTombstoneReplicationItem revisionTombstoneItem)
+                {
+                    if (SupportedFeatures.Replication.RevisionTombstonesWithId)
+                    {
+                        var docId = _documentInfoHelper.GetDocumentId(revisionTombstoneItem);
+                        if (docId != null)
+                        {
+                            batches[_parent.Context.GetShardNumberFor(context, docId)].Items.Add(item);
+                            continue;
+                        }
+                    }
```

Comment:
this already done under `GetShardNumberForReplicationItem`, no?

Reply:
Yes, but it won't work or maybe return the wrong result if `DocId `is null so I prefer checking it directly here